File: rot13.go
package main

import "fmt"

func Rot13(word string, num int) string{
	res := []rune{}
	for _,ch:= range word{
		if ch >='a'||ch<='z'{
			res = append(res, 'a'+(ch-'a'+rune(num)%26))
		}else if ch >='A'||ch<='Z'{
			res = append(res, 'A'+(ch-'A'+rune(num)%26))
		}else{
			res = append(res,ch)
		}

	}

	return string(res)
}

func main(){
	fmt.Println(Rot13("Hello",3))
}

File: rot13.py
"""implementation of rot13 in python"""


def rot13(word, num):
    res = []
    for ch in word:
        if 'a' <= ch <= 'z':
            res.append(chr((ord(ch)-ord('a')+num) % 26+ord('a')))
        elif 'A' <= ch <= 'Z':
            res.append(chr((ord(ch)-ord('A')+num) % 26+ord('A')))
        else:
            res.append(ch)
    return ''.join(res)


print(rot13('hello', 3))
